Rover/Log.cpp

```diff
@@ -38,46 +38,53 @@ void Rover::Log_Write_Attitude()
 // Write a range finder depth message
 void Rover::Log_Write_Depth()
 {
-    // only log depth on boats with working downward facing range finders
-    if (!rover.is_boat() || !rangefinder.has_data_orient(ROTATION_PITCH_270)) {
+    // only log depth on boats
+    if (!rover.is_boat()) {
         return;
     }
 
     // get position
     Location loc;
     IGNORE_RETURN(ahrs.get_position(loc));
 
-    // check if new sensor reading has arrived
-    uint32_t reading_ms = rangefinder.last_reading_ms(ROTATION_PITCH_270);
-    if (reading_ms == rangefinder_last_reading_ms) {
-        return;
-    }
-    rangefinder_last_reading_ms = reading_ms;
-
-    // get temperature
-    float temp_C;
-    if (!rangefinder.get_temp(ROTATION_PITCH_270, temp_C)) {
-        temp_C = 0.0f;
+    for (uint8_t i=0; i<RANGEFINDER_MAX_INSTANCES; i++) {
+        AP_RangeFinder_Backend *s = rangefinder.get_backend(i);
+        
+        if (s == nullptr || s->orientation() != ROTATION_PITCH_270 || !s->has_data()) {
+            continue;
+        }
+
+        // check if new sensor reading has arrived
+        const uint32_t reading_ms = s->last_reading_ms();
+        if (reading_ms == rangefinder_last_reading_ms[i]) {
+            continue;
+        }
+        rangefinder_last_reading_ms[i] = reading_ms;
+
+        float temp_C;
+        temp_C = s->get_temp(temp_C) ? temp_C : 0.0f; 
+
+        // @LoggerMessage: DPTH
+        // @Description: Depth messages on boats with downwards facing range finder
+        // @Field: TimeUS: Time since system startup
+        // @Field: Inst: Instance
+        // @Field: Lat: Latitude 
+        // @Field: Lng: Longitude   
+        // @Field: Depth: Depth as detected by the sensor
+        // @Field: Temp: Temperature
+
+        logger.Write("DPTH", "TimeUS,Inst,Lat,Lng,Depth,Temp",
+                            "s#DUmO", "F-GG00", "Q-LLff",
+                            AP_HAL::micros64(),
+                            i,
+                            loc.lat,
+                            loc.lng,
+                            (double)(rangefinder.distance_cm_orient(ROTATION_PITCH_270) * 0.01f),
```

Comment:
This is a bug here. We need to use the backend we retrieved since more than one can be rotation_pitch_270.
```suggestion
                            (double)(s->distance_cm() * 0.01f),
```

Reply:
Sorry, I forgot to update this. Fixed it now.